### ISSUE TITLE
  In the case of distributed training, the accuracy has not increased.

### ISSUE THREAD

**Issue by superccy** [dmlc/dgl#3507]
## ❓ Questions and Help
When using reddit dataset to train GCN and graphsage in distributed scenarios, the accuracy has not been improved, which is similar to that of the first epoch training. What is the reason?

**Comment by VoVAllen** [comment]
Hi,

What's the script you are using and what's your dataset?

**Comment by superccy** [comment]
> Hi,
> 
> What's the script you are using and what's your dataset?

The script is / dgl / examples / pytorch / graphsage / train_ dist.py. And the dataset is a reddit dataset built into DGL.

**Comment by VoVAllen** [comment]
Have you tried ogb product dataset also? Did you see the same learning pattern?
Also how many partition are you using?

**Comment by superccy** [comment]
> Have you tried ogb product dataset also? Did you see the same learning pattern? Also how many partition are you using?

Sorry, I haven't tried to test the OGB dataset yet. However, I tested the PubMed dataset without data division, that is, all data vertices are training data, and the training accuracy can be improved. I divided the data set into four partitions and used four Alibaba Cloud Servers for experiments.

**Comment by superccy** [comment]
> Have you tried ogb product dataset also? Did you see the same learning pattern? Also how many partition are you using?

I have written the specific details here： https://discuss.dgl.ai/t/in-the-case-of-distributed-training-the-accuracy-has-not-increased/2490/3.

**Comment by Rhett-Ying** [comment]
let's discuss in this channel: https://discuss.dgl.ai/t/in-the-case-of-distributed-training-the-accuracy-has-not-increased/2490
